File: langs/java/utils/Tester.java
public class Tester <T>{
  public static void main(String[] args){
    Tester<Integer> t = new Tester <>("Main Test");
    t.test("test-1",10, 10);
  }
  public final String title; private Colors c;
  Tester(String title){this.title = title; this.c = new Colors();}
  
  void test(String dsp, T act, T exp){
    String front = String.format("%s %s   %-25s", c.norm("@"),this.title, "disp is : "+ dsp); String res;
    if(act.equals(exp)){ res = String.format("%23s actual : %-10s => %s passed!", c.suc("Succes! "), act.toString(), c.suc("OK!"));}
    else{
      front += String.format("%28s\n", "** " + c.err("Fielid!!"));
      res = String.format("%20s %s\n", " ", "-- Expect : " + exp.toString()) + String.format("%22s %s", " ", "bat was : " + act.toString());
    }
    System.out.println(front + res);
  }
}

File: langs/ocaml/utils/tester.ml
#use "a:/PJ/SC/langs/OCaml/utils/print_util.ml"
#use "a:/PJ/SC/langs/OCaml/utils/to_string.ml"
#use "a:/PJ/SC/langs/OCaml/utils/colors.ml"

(* これは汎用のテストメソッドなので、テストは無理でもないよ！！*)
let zz_act ?(dsp = "now printing!!") name act exp = 
  let title = sprintf "%s %-10s  test name is  %-15s" (norm "@") name ("<" ^ (fine dsp) ^ ">") and sp = " " in
    let res = if act = exp
      then sprintf "%20s actual : %-20s => %s passed" (suc "Succes!") act (suc "OK!")
      else sprintf "%5s **%s \n %10s %s \n %12s %s" sp (err "Fielid!") sp ("-- Expect : " ^ exp) sp ("bat was : " ^ act)
	in printf "%s %s \n" title res

let tester_test title = 
  zz_act ~dsp:title "Tester Test" (sI 10) (sI 10);
  zz_act ~dsp:title "Tester Test" (sI 10) (sI 5)


(*
let () = tester_test "Teter test"
*)